textile/features.textile

```diff
@@ -552,9 +553,9 @@ h3(#realtime-connection). Connection
 *** @(RTN17f1)@ a @DISCONNECTED@ response with an @error.statusCode@ in the range @500 <= code <= 504@
 ** @(RTN17e)@ If the realtime client is connected to a fallback host endpoint, then for the duration that the transport is connected to that host, all HTTP requests, such as history or token requests, should be first attempted to the same datacenter the realtime connection is established with i.e. the same fallback host must be used as the default HTTP request host. If however the HTTP request against that fallback host fails, then the normal fallback host behavior should be followed attempting the request against another fallback host as described in "RSC15":#RSC15
 * @(RTN19)@ Transport state side effects - when a transport is disconnected for any reason:
-** @(RTN19a)@ Any @ProtocolMessage@ that is awaiting an @ACK@/@NACK@ on the old transport will not receive the @ACK@/@NACK@ on the new transport. The client library must therefore resend any @ProtocolMessage@ that is awaiting a @ACK@/@NACK@ to Ably in order to receive the expected @ACK@/@NACK@ for that message (subject to @RTN7c@/@RTN7d@). The Ably service is responsible for keeping track of messages, ignoring duplicates and responding with suitable @ACK@/@NACK@ messages
-*** @(RTN19a1)@ One possible implementation of this requirement would be to add any in-flight messages to the @RTL6c2@ connection-wide queue of messages that will be sent once the connection next becomes @CONNECTED@
-*** @(RTN19a2)@ In the case of an @RTN15c6@ successful resume, the @msgSerial@ of the reattempted @ProtocolMessage@s should remain the same as for the original attempt. In the case of an @RTN15c7@ failed resume, the message must be assigned a new @msgSerial@ from the SDK's internal counter.
+** @(RTN19a)@ Any @ProtocolMessage@ that is awaiting an @ACK@/@NACK@ on the old transport will not receive the @ACK@/@NACK@ on the new transport (failed resume as per @RTN15c7@). The client library must therefore resend any @ProtocolMessage@ that is awaiting a @ACK@/@NACK@ to Ably in order to receive the expected @ACK@/@NACK@ for that message (subject to @RTN7c@/@RTN7d@). The Ably service is responsible for keeping track of messages, ignoring duplicates and responding with suitable @ACK@/@NACK@ messages
```

Comment:
No, not just failed resume, rtn19a applies to successful resumes as well

Reply:
I am not getting. It says new transport, that means it will be a new connection id right? We only get new `connectionId` for a failed resume right? Okay, there can be a case where it's not a resume attempt as such 🤔.
I think spec should be modified so we have explicit understanding that, new transport means `new connectionId`. If it's same connectionId, then we don't need to add it to the queue, and it doesn't have anything to do with resume as such

Reply:
> I think spec should be modified so we have explicit understanding that, new transport means new connectionId.

It doesn't mean that. A transport is not an Ably Connection, it's the level below, eg a websocket (effectively tcp) connection. This isn't our own jargon, it's OSI terminology

Like -- RTN19a2 explains that you have to do something slightly different if the connectionId changes vs if it stays the same. If the whole of RTN19a only applied if the connectionId changed, that wouldn't make any sense.

> If it's same connectionId, then we don't need to add it to the queue, and it doesn't have anything to do with resume as such

You might think that. But then you'd read RTN19a and discover that you were wrong, the message needs to be resent on any new transport, not just a new connectionId. That's why RTN19a exists 🙂

Reply:
So, what happens if you have a `new connectionId` and `no error` in errorField. 
1. Successful resume means `same connectionid` and `no error`. 
2. Failed resume means `new connectionid` and `error`
For both of above cases, request should have `connectionkey` sent.
For new transport, there can be a case where no `connectionkey` is sent and new `connectionId` is received and there can a error or no error. So as per new transport ( new connectionId received ), we should append to the queue. So, we can ignore all resume related checks I feel

Reply:
I am confused here. I thought new transport clearly meant `new connctionId`? Because that's how ably maintains connectionState on the server right? Otherwise, there can always be a new raw transport when new connection is made and without `connectionId` we will not be able to detect if it's coming from same old device.

Reply:
What if you have a new raw transport and same connectionid? This means connection is resumed yet we won't be able to receive ack for sent messages 🤔. Clearly contradicts with `RTN19b`

Reply:
Okay, it seems new transport has nothing to do with maintaining connection state, or `connectionState` doesn't maintain pending ack/nack on the server. It just maintains pending messages.